convert2rhel/utils.py

```diff
@@ -560,4 +545,26 @@ def backup(self):
             loggerinst.warning("Can't access %s" % TMP_DIR)
 
 
+def check_readonly_mounts():
+    """
+    Mounting directly to /mnt/ is not in line with Unix FS (https://en.wikipedia.org/wiki/Unix_filesystem).
+    Having /mnt/ and /sys/ read-only causes the installation of the filesystem package to
+    fail (https://bugzilla.redhat.com/show_bug.cgi?id=1887513, https://github.com/oamg/convert2rhel/issues/123).
+    """
+    mounts = get_file_content('/proc/mounts', as_list=True)
+    for line in mounts:
+        _, mount_point, _, flags, _, _ = line.split()
+        flags = flags.split(',')
+        if mount_point not in ('/mnt', '/sys'):
+            continue
+        if 'ro' in flags:
+            if mount_point == '/mnt':
+                loggerinst.critical("Detected read-only mount to /mnt directory. This will cause the conversion to fail.\n"
+                                    "Mount to a subdirectory of /mnt to have read-write.")
+            else:  # /sys
+                loggerinst.critical("Detected read-only mount to /sys directory. This will cause the conversion to fail.\n"
+                                    "Mount in read-write mode before executing convert2rhel.")
+        loggerinst.debug("%s mount is not read-only." % mount_point)
```

Comment:
In the `mount` man page, they use "mount filesystem/device [on/at] a directory". So they are not consistent, but they do not use "mount filesystem/device to a directory". _Mount_ is a verb, and _mount point_ is the directory on/at which you mount a filesystem/device.

```suggestion
                loggerinst.critical("Detected /mnt as a read-only mount point. That would cause the conversion to fail.\n"
                                    "Use any subdirectory of /mnt as a mount point instead to have /mnt writeable.")
            else:  # /sys
                loggerinst.critical("Detected /mnt as a read-only mount point. That would cause the conversion to fail.\n"
                                    "Mount the directory in read-write mode before executing convert2rhel.")
        loggerinst.debug("%s mount point is not read-only." % mount_point)
```

Reply:
> That would cause the conversion to fail

Why past tense rather than present? IMO if we describe it in past tense here it indicates we're being nice to the user rather than simply stating we can't continue. That we're doing them a favor of stopping the conversion early

> Use any subdirectory of /mnt as a mount point instead to have /mnt writeable.

I find this rather complex rather, this would be better: "Mount to a subdirectory of /mnt to have a [read-write|writable] mount point."

> Mount the directory in read-write mode before executing convert2rhel

We're mounting to a directory in /mnt, not mounting a specific directory

Reply:
> Why past tense rather than present? IMO if we describe it in past tense here it indicates we're being nice to the user rather than simply stating we can't continue. That we're doing them a favor of stopping the conversion early

The "would" does not IMHO imply past tense here. It implies a possibility in the future. I.e. should we not stop you here to make amendments the conversion would fail.

> I find this rather complex rather, this would be better: "Mount to a subdirectory of /mnt to have a [read-write|writable] mount point."

That's indeed better, thanks.

> We're mounting to a directory in /mnt, not mounting a specific directory

Agreed. What about "Make sure this mount point is writeable before executing convert2rhel" to workaround it?

Reply:
> The "would" does not IMHO imply past tense here. It implies a possibility in the future. I.e. should we not stop you here to make amendments the conversion would fail.

Hm. I think we can skip it altogether actually. We don't need to tell the user it will fail, as we just stopped it - so it won't ever "fail".

What about doing:

> Stopping conversion due to read-only mount to /mnt directory
> Mount to a subdirectory of /mnt to have read-write.

---

> Agreed. What about "Make sure this mount point is writeable before executing convert2rhel" to workaround it?

I would go with "ensure" rather than "make sure":

> Ensure mount point is writable before executing convert2rhel

Reply:
> I would go with "ensure" rather than "make sure":

Sure.

The _"Mount to a subdirectory of /mnt to have read-write."_ sounds strange. Especially the "having read-write". What is read-write and why should I have it? :)
Also, as I wrote, the man page does not use _mount to_, but _mount on_ or _mount at_. So what about:

```
Stopping the conversion due to read-only mount at /mnt directory.
Mount at a subdirectory of /mnt instead to have /mnt writeable.
```

Reply:
Almost agree, I'd drop `instead` as it's an unneeded filler :)

Reply:
Fine with me.

Reply:
Something like this then

```suggestion
                loggerinst.critical("Stopping conversion due to read-only mount to /mnt directory.\n"
                                    "Mount at a subdirectory of /mnt to have /mnt writeable.")
            else:  # /sys
                loggerinst.critical("Stopping conversion due to read-only mount to /sys directory.\n"
                                    "Ensure mount point is writable before executing convert2rhel.")
        loggerinst.debug("%s mount point is not read-only." % mount_point)
```